bridge/slack/slack.go

```diff
@@ -310,6 +310,13 @@ func (b *Bslack) prepareMessageParameters(msg *config.Message) *slack.PostMessag
 	params.Username = msg.Username
 	params.LinkNames = 1 // replace mentions
 	params.IconURL = config.GetIconURL(msg, b.GetString(iconURLConfig))
+	msgFields := strings.Fields(msg.ParentID)
+	if len(msgFields) >= 2 {
+		msgProtocol, msgID := msgFields[0], msgFields[1]
+		if msgProtocol == "slack" {
+			params.ThreadTimestamp = msgID
+		}
+	}
```

Comment:
You can just do  (like you had in your original code), no need to doublecheck here, we should only be getting valid input.
```
params.ThreadTimestamp = msgFields[1]
```

Reply:
that was failing, I assumed because some messages are coming through without an ID set in this way

Reply:
And if there's an incorrect parentID, it doesn't just post to main channel, but loses the message completely. So resetting to "" at the start is important